plugins/web-worker-offloading/hooks.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Hook callback for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets configuration for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @link https://partytown.builder.io/configuration
+ *
+ * @return array{ debug?: bool, forward?: non-empty-string[], lib: non-empty-string, loadScriptsOnMainThread?: non-empty-string[], nonce?: non-empty-string } Configuration for Partytown.
+ */
+function wwo_get_configuration(): array {
+	$config = array(
+		'lib'     => wp_parse_url( plugin_dir_url( __FILE__ ), PHP_URL_PATH ) . 'build/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for Web Worker Offloading.
+	 *
+	 * @since n.e.x.t
+	 * @link https://partytown.builder.io/configuration
+	 *
+	 * @param array{ debug?: bool, forward?: non-empty-string[], lib: non-empty-string, loadScriptsOnMainThread?: non-empty-string[], nonce?: non-empty-string } $config Configuration for Partytown.
+	 */
+	return apply_filters( 'wwo_configuration', $config );
+}
+
+/**
+ * Initialize Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ */
+function wwo_init(): void {
+	$partytown_js = file_get_contents( __DIR__ . '/build/partytown.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
+	if ( false === $partytown_js ) {
+		return;
+	}
+
+	wp_register_script(
+		'web-worker-offloading',
+		'',
+		array(),
+		WEB_WORKER_OFFLOADING_VERSION,
+		array( 'in_footer' => false )
+	);
+
+	wp_add_inline_script(
+		'web-worker-offloading',
+		sprintf(
+			'window.partytown = %s;',
+			wp_json_encode( wwo_get_configuration() )
+		),
+		'before'
+	);
+
+	wp_add_inline_script( 'web-worker-offloading', $partytown_js );
+}
+add_action( 'wp_enqueue_scripts', 'wwo_init' );
+
+/**
+ * Helper function to get all scripts tags which has `web-worker-offloading` dependency.
+ *
+ * @since n.e.x.t
+ *
+ * @return string[] Array of script handles.
+ */
+function wwo_get_web_worker_offloading_handles(): array {
+	/**
+	 * Array of script handles which has `web-worker-offloading` dependency.
+	 *
+	 * @var string[] $web_worker_offloading_handles
+	 */
+	static $web_worker_offloading_handles = array();
+
+	if ( count( $web_worker_offloading_handles ) > 0 ) {
+		return $web_worker_offloading_handles;
+	}
```

Comment:
This makes me a bit nervous because what the scripts are printed in head, and then in the body a new script is registered for printing in the footer? Then it would never be picked up.

Reply:
Per the below suggestion, this entire block of code could be removed since it is only executed a couple times when the `print_scripts_array` filter applies.

```suggestion
	$web_worker_offloading_handles = array();
```